lib/mprintf.c

```diff
@@ -1008,7 +1008,7 @@ int curl_mvsnprintf(char *buffer, size_t maxlength, const char *format,
     /* we terminate this with a zero byte */
     if(info.max == info.length)
       /* we're at maximum, scrap the last letter */
-      info.buffer[-1] = 0;
+      info.buffer[maxlength - 1] = 0;
```

Comment:
`buffer` is incremented in the `addbyter` function so this is likely to introduce an out-of-boundary access. If you suspect this line is wrong, I would encourage you to also add some tests that show it/verify the fix.

Reply:
`buffer` pointer is not incremented but `info.buffer` which is a copy of it is. My guess since there's no description is either he meant `buffer` and was trying to refactor and/or he misinterpreted -1 as info.buffer[4294967295]. The logic is essentially if entire buffer written then `(*(info.buffer+(-1))) = 0` else `(*(info.buffer+(0))) = 0`. Or maybe something else is going on in dprintf_formatf because the branch is labeled seg_fault.2.

Reply:
It was simple a code review whilst chatting to @bagder on IRC this morning about what the issue might be - It was a theory (certainly not a definite cause) and I had missed where `info.buffer` was incremented as I didn't realise `addbyter` was passed as and argument.

I will close the PR ;-)